open-craft/opencraft
# [BB-2082] Deploy the simple theme using the theme configuration

Adds the changes to deploy the `simple-theme` using the theme configuration stored in the `theme_config` field. Also defines schema for the v0, v1 configuration, adds validators and unit tests. Also adds two new `SIMPLE_THEME_SKELETON_THEME_LEGACY*` variables for using a different branch for the v0 theming implementation.

**Testing instructions**:
This branch is already deployed on stage and there is a sandbox to test and verify the functionality (details in the task).
* Checkout the source branch of this PR.
* Configure the values for the `SIMPLE_THEME_SKELETON_THEME_*` variables.
* Create a new beta-test instance from the registration form.
* Verify that the simple theme configuration for the initial appserver is correct.
* Once the first appserver is provisioned, activate it and test the LMS theming and verify that the theme configuration has been applied correctly.
* Add valid theme configuration in the `theme_config` field of the instance. A valid configuration based on the variable mappings listed in the ticket is already available for usage or as reference.
* Check the generated simple theme configuration in the instance settings.
* Launch a new appserver for the instance.
* Once that is provisioned succesfully, check its LMS and verify that the theming is applied as configured in the `theme_config` field.

**Author notes and concerns**:
* The schema constraints are very lax and in the v1 schema all the theme variables are optional at the moment. This can be changed once we get a better idea of what those need to be. 
* The CI checks are failing for unrelated issues.
